h/security/predicates.py

```diff
@@ -143,12 +143,32 @@ def group_created_by_user(identity, context):
 
 
 @requires(authenticated_user, group_found)
-def group_has_user_as_member(identity, context):
-    # With detached groups like we have with the websocket, this doesn't work
-    # as SQLAlchemy does not consider them equal:
-    # return context.group in identity.user.groups
+def group_has_user_as_owner(identity, context):
+    return any(
+        owner.id == identity.user.id
+        for owner in context.group.get_members(GroupMembershipRoles.OWNER)
+    )
+
+
+@requires(authenticated_user, group_found)
+def group_has_user_as_admin(identity, context):
+    return any(
+        admin.id == identity.user.id
+        for admin in context.group.get_members(GroupMembershipRoles.ADMIN)
+    )
+
 
-    return any(user_group.id == context.group.id for user_group in identity.user.groups)
+@requires(authenticated_user, group_found)
+def group_has_user_as_moderator(identity, context):
+    return any(
+        moderator.id == identity.user.id
+        for moderator in context.group.get_members(GroupMembershipRoles.MODERATOR)
+    )
```

Comment:
Hard-coding the three roles owner, admin and moderator here, but given how this "predicates" system works I don't see a choice without refactoring it (I don't think the permission map can pass arguments to predicates).